src/api_helpers.jl

```diff
@@ -242,7 +242,7 @@ function h5s_get_simple_extent_dims(space_id)
     dims = Vector{hsize_t}(undef, n)
     maxdims = Vector{hsize_t}(undef, n)
     h5s_get_simple_extent_dims(space_id, dims, maxdims)
-    return dims, maxdims
+    return n, dims, maxdims
```

Comment:
@jmert It's not a huge performance hit to calcluate `n` again in `get_dims`, but since we have this information I figured it's best to just use it.

Reply:
I recall we might have discussed returning `n` a while ago when you wrote these `api_helpers`. Happy to keep as is and calculate `N = length(h5_dims)`, if you prefer otherwise.

Reply:
Yeah, my hesitation to returning it was/is that it deviates from the simple rule that the low-level API either follows the HDF5 docs (with the two auto-changes being `htri_t` -> throw on error or `Bool` and `herr_t` -> throw on error or `nothing`) or returns the documented out-args for the signature lacking those arguments.

I'm completely open to making the change here and elsewhere if the rule/transformation can be as clearly enumerated. (out args plus query-call return?)

Reply:
> Yeah, my hesitation to returning it was/is that it deviates from the simple rule that the low-level API either follows the HDF5 docs

Yeah, that's why I backtracked on these changes after thinking about it a little longer.  At times I felt like it might be nice to veer of this, but consistency is more important. 

I'm really liking the 3-levels of abstractions we currently have. (1) low-level, (2) convenience low-level wrappers and (3) high-level wrappers in the main source file.

I'm also thinking we should consolidate the naming of this function and other here to adhere to `h5s_get_simple_extent_dims` -> `get_simple_extent_dims`. I.e. at the highest level, remove the `h5(*)_` prefix.